distributed/worker.py

```diff
@@ -2129,29 +2128,24 @@ def transition_cancelled_error(
         *,
         stimulus_id: str,
     ) -> RecsInstrs:
-        recs: Recs = {}
-        instructions: Instructions = []
-        if ts._previous == "executing":
-            recs, instructions = self.transition_executing_error(
-                ts,
-                exception,
-                traceback,
-                exception_text,
-                traceback_text,
-                stimulus_id=stimulus_id,
-            )
-        elif ts._previous == "flight":
-            recs, instructions = self.transition_flight_error(
-                ts,
-                exception,
-                traceback,
-                exception_text,
-                traceback_text,
-                stimulus_id=stimulus_id,
-            )
-        if ts._next:
-            recs[ts] = ts._next
-        return recs, instructions
+        assert ts._previous == "executing" or ts.key in self.long_running
```

Comment:
Isn't this equivalent?
```suggestion
        assert ts._previous in ("executing", "long-running")
```

Reply:
This has been used in other places of the code. I think it should be equivalent, I'll keep it like this for now